src/core/MRMedia.js

```diff
@@ -0,0 +1,242 @@
+import * as THREE from 'three';
+
+import { MRDivEntity } from 'mrjs/core/MRDivEntity';
+
+import { mrjsUtils } from 'mrjs';
+
+/**
+ * @class MRMedia
+ * @classdesc Base html media entity represented in 3D space. `mr-media`
+ * @augments MRDivEntity
+ */
+export class MRMedia extends MRDivEntity {
+    /**
+     * @class
+     * @description Constructs a base media entity using a UIPlane and other 3D elements as necessary.
+     */
+    constructor() {
+        super();
+        this.attachShadow({ mode: 'open' });
+
+        // Create the object3D. Dont need default value for geometry
+        // until the connected call since this will get overwritten anyways.
+        let material = new THREE.MeshStandardMaterial({
+            side: 0,
+            transparent: true,
+        });
+        // Object3D for MRMedia (mrimage,mrvideo,etc) is the actual image/video/etc itself in 3D space
+        this.object3D = new THREE.Mesh(undefined, material);
+        this.object3D.receiveShadow = true;
+        this.object3D.renderOrder = 3;
+        this.object3D.name = 'media';
+
+        // the media to be filled out.
+        // for ex: document.createElement('video') or document.createElement('img');
+        this.media = null;
+
+        // This is a reference to the texture that is used as part of the
+        // threejs material. Separating it out for easier updating after it is loaded.
+        // The texture is filled-in in the connected function.
+        this.texture = null;
+
+        // This is used to aid in the formatting for certain object-fit setups
+        // ex: contain, scale-down
+        this.subMediaMesh = null;
+    }
+
+    /**
+     * @function
+     * @description Calculates the width of the img based on the img tag in the shadow root
+     * @returns {number} - the resolved width
+     */
+    get width() {
+        let width = this.objectFitDimensions?.width;
+        return width > 0 ? width : super.width;
+    }
+
+    /**
+     * @function
+     * @description Calculates the height of the img based on the img tag in the shadow root
+     * @returns {number} - the resolved height
+     */
+    get height() {
+        let height = this.objectFitDimensions?.height;
+        return height > 0 ? height : super.height;
+    }
+
+    get mediaWidth() {
+        // to be filled in in children
```

Comment:
Nit, not blocking issue: Do we want to throw an error here? We may be able to more easily detect in cases if we forgot to override this method in child classes.

Reply:
√